src/agent/agent.c

```diff
@@ -1939,12 +1939,16 @@ static int agent_method_switch_controller(sd_bus_message *m, void *userdata, UNU
                                 "Failed to switch controller because already connected to the controller");
         }
 
-        if (!agent_set_controller_address(agent, dbus_address)) {
+        if (!agent_set_controller_address(agent, dbus_address) ||
```

Comment:
I'd extract these two to some function like `agent_update_controller_address`, at this point it is completely unclear why are we doing these two.

We need to improve the naming here though. Separately, of course

Reply:
> I'd extract these two to some function like agent_update_controller_address, at this point it is completely unclear why are we updating these two.

I added an inline-comment to explain the reasoning behind this since its a bit complex.

> We need to improve the naming here though. Separately, of course

Yes, the naming is a bit irritating (result from history). I applied the name changes directly in this PR (separate commit) so this gets a bit clearer.